test/unit/app/controllers/permissions/permissions-controller-test.js

```diff
@@ -747,14 +749,14 @@ describe('permissions controller', function () {
 
       assert.deepEqual(
         Object.keys(permissionsHistory[ORIGINS.a].eth_accounts.accounts),
-        ACCOUNT_ARRAYS.a,
+        ACCOUNTS.a.permitted,
         'should have expected eth_accounts entry accounts'
       )
 
       // notification should also have been sent
       assert.deepEqual(
         notifications[ORIGINS.a][0],
-        NOTIFICATIONS.newAccounts(ACCOUNT_ARRAYS.a),
+        NOTIFICATIONS.newAccounts(ACCOUNTS.a.permitted),
```

Comment:
Why is this not `ACCOUNTS.a.primary`?

Reply:
legacyExposeAccounts doesn't currently have the primary account limit enforced (seems to be bypassing caveats in general) -- and a entry is made to the permissionLog that is the direct input to the legacyExposeAccounts method. 

Should I do something in this legacy method to restrict to primary as well?

Reply:
`legacyExposeAccounts` is likely to be deleted by @whymarrh in the near future I suspect. Once [this card](https://trello.com/c/o8ZRtfa9/11-manually-connect-my-current-account-to-the-current-site) is completed, it won't be used anymore.

Reply:
I pushed up the rest of the fixes, this one is still strange tho.. not sure if we should merge this as is or restrict the account access in the legacy method -- even if it will be removed

Reply:
Ah, that's fair. But, `legacyExposeAccounts` does not bypass caveats! It's merely the notification that does. It calls `notifyDomain` inside a Promise constructor callback. `notifyDomain` should be called with `result: await this.getAccounts(origin)` _after_ awaiting the Promise.

Reply:
ah yeah, this is only for setting permissions not actually accessing accounts. my bad. So we would expect permission history to show all the accounts added.

Reply:
Yeah, and the notification sent at the moment is plain wrong, possibly even without the `primaryAccountOnly` caveat.

Reply:
This appears to have been done!